python/ray/train/torch/config.py

```diff
@@ -178,27 +177,9 @@ def on_shutdown(self, worker_group: WorkerGroup, backend_config: TorchConfig):
         )
 
     @staticmethod
-    def encode_data(data_dict: Dict) -> EncodedData:
-        """Special handling for moving model from worker to driver."""
+    def _get_checkpoint_class(data_dict: Dict) -> Type[TorchCheckpoint]:
```

Comment:
Instead of introducing this `checkpoint_class` concept and plumbing this through, could we just do something like this instead?

```
class TorchBackend:

def encode_data(data_dict):
    return TorchCheckpoint.from_dict(data_dict)

def decode_data(data):
    return data.to_dict()
```

Reply:
I don't think we should encode and decode data in the backends. This requires us to convert checkpoints to dict, when the point of this PR is to avoid this. Furthermore, the encoding/decoding issues are not Train specific - if you pass a dict Torch checkpoint around, it will most likely cause serialization issues too, even if it wasn't created using Train. Therefore, I think it makes sense that any special logic necessary to serialize a checkpoint is in the checkpoint class itself.